diff --git a/CMakeLists.txt b/CMakeLists.txt
index 68973eb7..194f8954 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -200,8 +200,8 @@ endif()
 # EIGEN
 FetchContent_Declare(
     eigen
-    URL "http://bitbucket.org/eigen/eigen/get/3.3.7.tar.gz"
-    URL_HASH MD5=f2a417d083fe8ca4b8ed2bc613d20f07)
+    URL "https://gitlab.com/libeigen/eigen/-/archive/3.3.7/eigen-3.3.7.tar.gz"
+    URL_HASH MD5=9e30f67e8531477de4117506fe44669b)
 FetchContent_GetProperties(eigen)
 if(NOT eigen_POPULATED)
     FetchContent_Populate(eigen)
